solr/core/build.gradle

```diff
@@ -138,6 +138,8 @@ dependencies {
 
   testImplementation ('com.sun.jersey:jersey-servlet') { transitive = false }
 
+  // required for instantiating a Zookeeper server in tests or embedded
+  runtimeOnly ('org.xerial.snappy:snappy-java') { transitive = false }
```

Comment:
I would exclude any dependencies this has explicitly (not by just saying all of transitive dependencies should be omitted).

Reply:
I'm not sure there are any. There's probably a nifty way to make Gradle _tell_ me what the dependencies are, and "gradlew dependencies" seems to work for most other packages, but even if I rebuild with transitive = true, then run "gradlew dependencies", there are no dependencies shown for snappy. Hints?

Reply:
gradlew dependencies is the way to go as it shows you exactly which dependencies it includes (for each configuration). This means that compilation-time dependencies may be different than runtime dependencies, for example. For a given configuration it will show you the subtree of dependencies for a given entry. If snappy doesn't have any sub-dependencies then the subtree is empty.

Alternatively, you can take a look at the pom file directly. :)

https://repo1.maven.org/maven2/org/xerial/snappy/snappy-java/1.1.7.6/snappy-java-1.1.7.6.pom

I only see test dependencies and "provided" which isn't included transitively. So you can remove that "transitive = false" block as it doesn't do anything for snappy. It is weird to me that this dependency isn't included by zookeeper itself (it should be!). Perhaps it's because zookeeper is declared with transitive=false. If so then a much better way to solve this would be to remove the explicit snappy dependency and just include it transitively with zookeeper.